tf2onnx/tfonnx.py

```diff
@@ -185,18 +187,105 @@ def direct_op(ctx, node, name, args):
 def identity_op(ctx, node, name, args):
     """Identity."""
     if node.inputs[0].is_const():
-        # if identity has a const as input, remove it
+        # if identity has a const as input, remove it if it is not an model output
         input_name = node.input[0]
         output_name = node.output[0]
         for n in ctx.get_nodes():
             for i, parent_name in enumerate(n.input):
                 if parent_name == output_name:
                     n.input[i] = input_name
-        return None
+        if not ctx.is_model_output(output_name):
+            return None
+
     ctx.copy_shape(node.input[0], node.output[0])
     return node
 
 
+def range_op(ctx, node, name, args):
+    """Range."""
+    start_node = node.inputs[0]
+    limit_node = node.inputs[1]
+    delta_node = node.inputs[2]
+
+    output_name = node.output[0]
+    base_name = utils.make_name(node.name)
+
+    if all(i.is_const() for i in node.inputs):
+        # Generate range as const if possible, start/limit/delta are all scalars with same type
+        start = start_node.get_tensor()
+        limit = limit_node.get_tensor()
+        delta = delta_node.get_tensor()
+        val = np.arange(start, limit, delta, dtype=start.dtype)
+        const_range = ctx.make_const(base_name, val)
+        ctx.replace_all_inputs(ctx.get_nodes(), output_name, const_range.name)
+        return None
+
+    nodes = []
+
+    start_name = start_node.output[0]
+    limit_name = limit_node.output[0]
+    delta_name = delta_node.output[0]
+
+    # trip_count
+    diff_name = "{}_diff".format(base_name)
+    nodes.append(Node(helper.make_node("Sub", [limit_name, start_name], [diff_name], name=diff_name), ctx))
```

Comment:
for output id, usually we use port_name(diff_name)

Reply:
sure